proposal/README.md

```diff
@@ -0,0 +1,239 @@
+
+### Okay, so there might be a theoretical equivalence, but are there any *practical* security differences with assumeutxo (vs. assumevalid)?
+
+Yes, there is one practical security difference. Currently, if I wanted to trick someone into thinking I had coins that I don't on the honest network, I'd have to
+
+- get them to start bitcoind with a bad `-assumevalid=` parameter,
+- isolate their node from the honest network to prevent them from seeing the
+  most-PoW headers chain, and
+- build a PoW-compatible chain that includes the existing [checkpoints](https://github.com/bitcoin/bitcoin/blob/91a25d1e711bfc0617027eee18b9777ff368d6b9/src/chainparams.cpp#L146-L160).
+
+This obviously involves a bit of effort since the attacker needs to generate a chain of blocks along with the requisite PoW.
+
+However, with assumeutxo if I can get the user to accept a malicious `assumeutxo` commitment value, most of the work is done. Modifying and serializing a false UTXO snapshot is quite easy -- no proof of work necessary.
+
+### That sounds really bad - so all an attacker has to do is get a user to accept a bad assumeutxo value and feed them a poisoned snapshot?
+
+Yes, that's all it takes.
+
+As a result, the assumeutxo value will be embedded in the source code and we will not build a mechanism that enables specification of the `assumeutxo` value via commandline; the practical risk is just too high. If users prefer to specify an alternate value (not recommended), they can modify the source code and recompile.
+
+The assumeutxo value will live in the source code; recall that if an attacker has means to affect the source code used to build a binary, they can already do anything they want.
+
+### It seems like this is an argument for including a commitment to the assumeutxo value in a place where it could be enforced by consensus, say in the block headers. Should we do that?
+
+Maybe in time, but not at the moment. Before we gain practical experience with use of UTXO snapshots, we don't know what the right commitment structure is. Making consensus changes is a very expensive process and should not be done until we're absolutely sure of what we want to commit to.
+
+Down the road we may introduce such a commitment into a consensus-critical place, but for now we should design assumeutxo to be secure without the assumption that we will.
+
+### Do you perform any extra validation on a loaded snapshot besides comparing its hash to the `assumeutxo` value?
+
+Yes. After the UTXO snapshot is loaded and the chain syncs to the network tip, we begin an initial block download process in the background using separate data structures (i.e. a separate chainstate). This background IBD will download and validate all blocks up to the last block assumed valid by the snapshot (i.e. the "base" of the snapshot).
```

Comment:
I propose that nodes should ask *all* their seeder connections to validate the UTXO snapshot hash. That way, if the hash is wrong (either because the software was maliciously edited, or more likely [but still pretty unlikely], a mistake was made), any one of the node's peers can inform them their hash is incorrect. At that point, the node will have reason to distrust the assumeutxo hash and can decide to go through the full process of building their own UTXO hash. Once phase 3 is put in place, even this scenario wouldn't actually require downloading and verifying the whole chain since the UTXO hash can be grabbed directly from the longest valid chain.